tket/tests/Circuit/test_Boxes.cpp

```diff
@@ -989,6 +989,178 @@ SCENARIO("Checking equality", "[boxes]") {
+    permutation[{0, 0}] = {1, 1};
+    permutation[{1, 1}] = {0, 0};
+    permutation[{0, 1}] = {1, 0};
+    permutation[{1, 0}] = {0, 1};
+    ToffoliBox tb(permutation);
+
+    Circuit comparison_circuit(2);
+    comparison_circuit.add_op<unsigned>(OpType::X, {1});
+    comparison_circuit.add_op<unsigned>(OpType::CnX, {1, 0});
+    comparison_circuit.add_op<unsigned>(OpType::X, {1});
+    comparison_circuit.add_op<unsigned>(OpType::CnX, {0, 1});
+    comparison_circuit.add_op<unsigned>(OpType::X, {1});
+    comparison_circuit.add_op<unsigned>(OpType::CnX, {1, 0});
+    comparison_circuit.add_op<unsigned>(OpType::X, {1});
+    comparison_circuit.add_op<unsigned>(OpType::CnX, {1, 0});
+    comparison_circuit.add_op<unsigned>(OpType::CnX, {0, 1});
+    comparison_circuit.add_op<unsigned>(OpType::CnX, {1, 0});
+
+    const auto optimised_unitary = tket_sim::get_unitary(*tb.to_circuit());
+    const auto basic_unitary = tket_sim::get_unitary(comparison_circuit);
+
+    CHECK(optimised_unitary.isApprox(basic_unitary));
+  }
+  GIVEN("Two qubit basis state permutation, one four states cycle") {
+    std::map<std::vector<bool>, std::vector<bool>> permutation;
+    permutation[{0, 0}] = {1, 1};
+    permutation[{1, 1}] = {0, 1};
+    permutation[{0, 1}] = {1, 0};
+    permutation[{1, 0}] = {0, 0};
+    ToffoliBox tb(permutation);
+
+    Circuit circ = *tb.to_circuit();
+
+    Circuit comparison_circuit(2);
+    comparison_circuit.add_op<unsigned>(OpType::X, {1});
+    comparison_circuit.add_op<unsigned>(OpType::CnX, {1, 0});
+    comparison_circuit.add_op<unsigned>(OpType::X, {1});
+    comparison_circuit.add_op<unsigned>(OpType::CnX, {0, 1});
+    comparison_circuit.add_op<unsigned>(OpType::X, {1});
+    comparison_circuit.add_op<unsigned>(OpType::CnX, {1, 0});
+    comparison_circuit.add_op<unsigned>(OpType::X, {0});
+    comparison_circuit.add_op<unsigned>(OpType::X, {1});
+    comparison_circuit.add_op<unsigned>(OpType::CnX, {0, 1});
+    comparison_circuit.add_op<unsigned>(OpType::X, {0});
+    comparison_circuit.add_op<unsigned>(OpType::X, {1});
+    comparison_circuit.add_op<unsigned>(OpType::CnX, {1, 0});
+    comparison_circuit.add_op<unsigned>(OpType::X, {1});
+
+    const auto optimised_unitary = tket_sim::get_unitary(circ);
+    const auto basic_unitary = tket_sim::get_unitary(comparison_circuit);
+    REQUIRE(circ.count_gates(OpType::CnX) == 3);
+
+    CHECK(optimised_unitary.isApprox(basic_unitary));
+  }
+  GIVEN("Three qubit basis state permutation, one four state cycle") {
+    std::map<std::vector<bool>, std::vector<bool>> permutation;
+    permutation[{0, 0, 1}] = {1, 1, 0};
+    permutation[{1, 1, 0}] = {0, 1, 0};
+    permutation[{0, 1, 0}] = {1, 0, 1};
+    permutation[{1, 0, 1}] = {0, 0, 1};
+
+    ToffoliBox tb(permutation);
+    Circuit circ = *tb.to_circuit();
+    const auto matrix = tket_sim::get_unitary(circ);
+
+    std::map<std::vector<bool>, std::vector<bool>> permutation1;
```

Comment:
Can't we just reuse `permutation`?

Reply:
I've removed this test anyway, but the current constructor consumes the permutation.

Reply:
Yup, I'd missed that.

Reply:
(See foot-gun passim!)